### DISCUSSION TITLE
Need help getting Model variables in registerMediaCollections

### DISCUSSION THREAD

**Discussion by AshPowell** [spatie/laravel-medialibrary#2056] [Q&A]
Hi, I feel like I'm missing something obvious here but I can't seem to query the current model variables inside of registerMediaCollections - not sure if the model isn't booted yet or something like that.

Here's my code - 
`    public function registerMediaCollections(): void
    {
        $filesAllowed = $this->plan->code != 'free' ? 5 : 1;

        $this->addMediaCollection('images')
            ->useDisk('s3')
            ->onlyKeepLatest($filesAllowed)
            ->registerMediaConversions(function (Media $media) {
                $this
                    ->addMediaConversion('thumb')
                    ->width(497)
                    ->optimize();
            });
    }`

Basically wanting to query if the user is on x plan then allow 5 images in the collection, else just allow one.

Thanks for any help - and apologies in advance I know it's something obvious but can't put my finger on it! 😂
